atlite/datasets/cmip.py

```diff
@@ -0,0 +1,259 @@
+    columns as 'lat' and 'lon'.
+
+    CMIP specifics; shif the longitude from 0..360 to -180..180. In addition
+    CMIP sometimes specify the time in the center of the output intervall this shifted
+    to the beginning.
+    """
+    ds = ds.assign_coords(lon=(((ds.lon + 180) % 360) - 180))
+    ds.lon.attrs["valid_max"] = 180
+    ds.lon.attrs["valid_min"] = -180
+    ds = ds.sortby("lon")
+
+    ds = ds.rename({"lon": "x", "lat": "y"})
+    # round coords since cds coords are float32 which would lead to mismatches
+    # ds = ds.assign_coords(
+    #     x=np.round(ds.x.astype(float), 5), y=np.round(ds.y.astype(float), 5)
+    # )
+
+    ds = maybe_swap_spatial_dims(ds)
+    if add_lon_lat:
+        ds = ds.assign_coords(lon=ds.coords["x"], lat=ds.coords["y"])
+    if add_ctime:
+        ds = ds.assign_coords(ctime=ds.coords["time"])
+
+    # shift averaged data to beginning of bin
+    ds = ds.assign_coords(time=ds.coords["time"].dt.floor(dt))
+
+    return ds
+
+
+def get_data(cutout, feature, tmpdir, lock=None, **creation_parameters):
+    """
+    Retrieve data from the ESGF CMIP database.
+
+    This front-end function downloads data for a specific feature and formats
+    it to match the given Cutout.
+
+    Parameters
+    ----------
+    cutout : atlite.Cutout
+    feature : str
+        Name of the feature data to retrieve. Must be in
+        `atlite.datasets.era5.features`
+    tmpdir : str/Path
+        Directory where the temporary netcdf files are stored.
+    **creation_parameters :
+        Additional keyword arguments. The only effective argument is 'sanitize'
+        (default True) which sets sanitization of the data on or off.
+
+    Returns
+    -------
+    xarray.Dataset
+        Dataset of dask arrays of the retrieved variables.
+
+    """
+    coords = cutout.coords
+
+    # sanitize = creation_parameters.get("sanitize", True)
+
+    if cutout.esgf_params == None:
+        with open(CMIP_SETUP_FILE, "r") as f:
+            cmip_params = yaml.safe_load(f)
+
+        model = creation_parameters.get("model")
+        if model:
+            try:
+                esgf_params = cmip_params[model]
+            except:
+                KeyError(f"{model} not reconized, update cmip.yml")
+        else:
+            raise (ValueError("Model not specified"))
+    else:
+        esgf_params = cutout.esgf_params
+    if esgf_params.get("frequency") == None:
+        if cutout.dt == "H":
+            freq = "h"
+        elif cutout.dt == "3H":
+            freq = "3hr"
+        elif cutout.dt == "6H":
+            freq = "6hr"
+        elif cutout.dt == "D":
+            freq = "day"
+        elif cutout.dt == "M":
+            freq = "mon"
+        elif cutout.dt == "Y":
+            freq = "year"
+        else:
+            raise (ValueError(f"{cutout.dt} not valid time frequency in CMIP"))
+
+    retrieval_params = {"chunks": cutout.chunks, "tmpdir": tmpdir, "lock": lock}
+
+    func = globals().get(f"get_data_{feature}")
```

Comment:
I am not sure if this is the preferred way to execute the function... @euronion any pythonic concerns? From my intuition I would prefer eval(...) but this might also not be the best option.

Reply:
I used the same way as what was done in the era5.py . Personally, I'm not that familiar with that syntax, so please enlighten me.

Reply:
alright let's keep it like this